plugins/azure-functions/src/api/AzureFunctionsBackendClient.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AzureFunctionsApi } from './AzureFunctionsApi';
+import { FunctionsListResponse } from '@backstage/plugin-azure-functions-common';
+import { DiscoveryApi, IdentityApi } from '@backstage/core-plugin-api';
+
+/** @public */
+export class AzureFunctionsBackendClient implements AzureFunctionsApi {
+  private readonly identityApi: IdentityApi;
+  private readonly discoveryApi: DiscoveryApi;
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    identityApi: IdentityApi;
+  }) {
+    this.discoveryApi = options.discoveryApi;
+    this.identityApi = options.identityApi;
+  }
+
+  async list({
+    functionName,
+  }: {
+    functionName: string;
+  }): Promise<FunctionsListResponse> {
+    try {
+      const url = `${await this.discoveryApi.getBaseUrl(
+        'azure-functions',
+      )}/list/${functionName}`;
+      const { token: idToken } = await this.identityApi.getCredentials();
```

Comment:
If using as a Bearer token, I think the token will be an access token, not an id token.
```suggestion
      const { token: accessToken} = await this.identityApi.getCredentials();
```